plugins/genocide/genocide.cpp

```diff
@@ -53,64 +66,67 @@ void genocide::Event(bz_EventData *eventData)
-        break;
-    }
-
-    case bz_eFlagGrabbedEvent:
-    {
-        /*   bz_FlagGrabbedEventData_V1* fge = (bz_FlagGrabbedEventData_V1*)eventData;*/
-        break;
-    }
-
-    case bz_eFlagDroppedEvent:
-    {
-        /* bz_FlagDroppedEventData_V1* fde = (bz_FlagDroppedEventData_V1*)eventData;*/
-        break;
-    }
-
-    case bz_eShotFiredEvent:
+    case bz_ePlayerDieEvent:
     {
-        bz_ShotFiredEventData_V1* sfed = (bz_ShotFiredEventData_V1*)eventData;
-
-        bz_BasePlayerRecord *playerRecord = bz_getPlayerByIndex(sfed->playerID);
-        if (!playerRecord)
+        bz_PlayerDieEventData_V1 *dieData = (bz_PlayerDieEventData_V1*)eventData;
+        //if its not a genocide kill, dont care
+        if (dieData->flagKilledWith != "G")
+            break;
+        // if the tank killed is a rogue and we don't have rogue as team enabled
+        if (!rogueAsTeam && dieData->team == eRogueTeam)
+            break;
+        // option to disallow genocide from being trigger by selfkills
+        if (disableSuicide && dieData->killerID == dieData->playerID)
             break;
 
-        bz_freePlayerRecord(playerRecord);
-        break;
-    }
+        // if we pass options, proceed to kill tanks of specified team
+        bz_APIIntList *playerList = bz_newIntList();
 
-    case bz_ePlayerDieEvent:
-    {
-        bz_PlayerDieEventData_V1* deed = (bz_PlayerDieEventData_V1*)eventData;
-        bz_ApiString& flag = deed->flagKilledWith;
+        bz_getPlayerIndexList(playerList);
 
-        if (flag == "G")
+        if (dieData->team == eRogueTeam)
         {
-            // if the tank killed was a rogue, kill all rogues.
-            bz_APIIntList *playerList = bz_newIntList();
+            for (unsigned int i = 0; i < playerList->size(); i++)
+            {
+                int targetID = (*playerList)[i];
+                bz_BasePlayerRecord *playRec = bz_getPlayerByIndex (targetID);
+                if (!playRec)
+                    continue;
 
-            bz_getPlayerIndexList(playerList);
+                // the sucker is a spawned rogue, kill him.  This generates another death event,
+                // so if you kill another rogue with geno while you are a rogue you end up dead too.
+                // and you get both messages (victim and be careful)
+                if (playRec->spawned && playRec->team == eRogueTeam)
+                {
+                    bz_killPlayer(targetID, false, dieData->killerID, "G");
+                    bz_sendTextMessage(BZ_SERVER, targetID, "You were a victim of Rogue Genocide");
 
+                    // oops, I ended up killing myself (directly or indirectly) with Genocide!
+                    if (targetID == dieData->killerID)
+                        bz_sendTextMessage(BZ_SERVER, targetID, "You should be more careful with Genocide!");
+                }
+
+                bz_freePlayerRecord(playRec);
+            }
+        }
```

Comment:
- These are identical loops, there's no need to split them for rogues only; combine the two.
- Is there a reason `eGenocideEffect` doesn't work for rogue tanks? Skimming through the source code, there doesn't seem to be anything special server-side or client-side disabling `eGenocideEffect` from affecting rogues. Can you test this behavior to make sure `eGenocideEffect` works or doesn't work?
- The "be careful message" shouldn't just apply to rogue genocide, that should apply to genocide in general, but should only appear if you're **not** rogue.